core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -1821,5 +1826,39 @@ private Flags() {}
     // to ensure that all static variables defined beforehand are initialized.
     static {
         FlagsLoaded.set();
+
+        if (warnNettyVersions()) {
```

Comment:
The Netty version warning block was moved here to avoid circular dependencies between `Flags` and `TransportTypeProvider`.